### ISSUE TITLE
Migration linter gives false negative when changing NULL/DEFAULT together

### ISSUE THREAD

**Issue by tildedave** [3YOURMIND/django-migration-linter#115]
Hi, we have a migration that was generated as passing the migration-linter (version 2.2.2) but is actually backwards incompatible.

```
ALTER TABLE “T” ALTER COLUMN “C” SET DEFAULT ‘’;
UPDATE “T” SET “C” = ‘’ WHERE “C” IS NULL;
ALTER TABLE “T” ALTER COLUMN “C” SET NOT NULL;
ALTER TABLE “T” ALTER COLUMN “C” DROP DEFAULT;
```

This is backwards incompatible as older code will not be able to create new records in the "T" table.  
  However the code [here](https://github.com/3YOURMIND/django-migration-linter/blob/master/django_migration_linter/sql_analyser/base.py#L9-L14
) flags it as okay since there is a "SET DEFAULT".  (It doesn't also notice the "DROP DEFAULT".)

I think a reasonable fix for this would be to also ensure that there is no "DROP DEFAULT" in the list of SQL statements.

**Comment by David-Wobrock** [comment]
Hi @tildedave thanks for reporting the issue.

Indeed, simply dropping the default on an existing column should be considered backward incompatible by the linter.
I'll try submitting a PR in the next days/weeks if I find the time, any PR would also be welcome :)

**Comment by David-Wobrock** [comment]
That is even the basic case that we actually want to catch :thinking:  A pity we missed it.
=> making a existing column NOT NULL with a default value through Django - but since Django only uses the DB default to fill existing columns but keeps the "real" default at an application level - it should be backward incompatible.

Good catch!

The PR https://github.com/3YOURMIND/django-migration-linter/pull/117 should fix the issue

**Comment by tildedave** [comment]
Oh thank you!  I was going to get to this ... but then work happened. 😄 Much appreciated.

**Comment by David-Wobrock** [comment]
> Oh thank you! I was going to get to this ... but then work happened. smile Much appreciated.

Sorry :sweat_smile: 
Don't hesitate to assign yourself on the issue next time, I don't steal the task :P 

But issue should be fixed in the linter version 2.3.0
